src/api/clients/singleton.ts

```diff
@@ -0,0 +1,63 @@
+import axios from "axios";
+import APIClientBase from "./base";
+import CONFIG from "@/config";
+import { APIError, APIRequestFailed } from "../index";
+import { getCookie } from "@/api/helpers";
+
+export default class SingletonAPIClient extends APIClientBase {
+  private static instance: SingletonAPIClient;
+
+  private constructor() {
+    const axiosInstance = axios.create({
+      baseURL: CONFIG.BASE_URL,
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+      withCredentials: true,
+    });
+
+    super(axiosInstance);
+
+    this.getRawAxiosInstance().interceptors.response.use(
+      (response) => response,
```

Comment:
We don't need the full response, we just need the data


```suggestion
      (response) => response.data,
```